src/aiida_quantumespresso/calculations/functions/seekpath_structure_analysis.py

```diff
@@ -28,4 +30,47 @@ def seekpath_structure_analysis(structure, **kwargs):
     # All keyword arugments should be `Data` node instances of base type and so should have the `.value` attribute
     unwrapped_kwargs = {key: node.value for key, node in kwargs.items() if isinstance(node, Data)}
 
-    return get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+    result = get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+
+    # If the input structure was a HubbardStructureData, update the primitive structure with Hubbard parameters
+    if isinstance(structure, HubbardStructureData):
+        update_structure_with_hubbard(structure, result)
+
+    return result
+
+
+def update_structure_with_hubbard(structure, result):
```

Comment:
But i am a lil puzzle from this example you give, you have different kinds to differentiate ,` V	Ni-3d	N-2p	2	45	1.1805 `, and `V	Ni-3d	N-2p	1	3	1.034` ??

Reply:
I guess for V, we will need to check distancee between index of atoms , and them compare and find the distance of the Hubbard(inital) to find which atoms in the primitive have the same distance to assign the new  V, and in this case we shouldnt use initialize_intersites_hubbard, since the Hubbard you provide in your example considers a different V depending on the distance

Reply:
> I guess for V, we will need to check distancee between index of atoms , and them compare and find the distance of the Hubbard(inital) to find which atoms in the primitive have the same distance to assign the new V

Would the distance between sites be sufficient? Can't there be two pairs of atoms that have the same distance but a different intersite V? In that case we'd have to check that the primitivization respects each V pair, not just within the unit cell using the atom index, but also the neighbors via the translations. 

>  and in this case we shouldnt use initialize_intersites_hubbard, since the Hubbard you provide in your example considers a different V depending on the distance

Indeed, `append_hubbard_parameter` would allow you to specify the atom/site indices and translation.